pkg/skaffold/build/local/local.go

```diff
@@ -89,36 +90,50 @@ func (b *Builder) buildArtifact(ctx context.Context, out io.Writer, a *latest.Ar
 	return build.TagWithImageID(ctx, tag, imageID, b.localDocker)
 }
 
-func (b *Builder) runBuildForArtifact(ctx context.Context, out io.Writer, a *latest.Artifact, tag string) (string, error) {
-	if !b.pushImages {
-		// All of the builders will rely on a local Docker:
-		// + Either to build the image,
-		// + Or to docker load it.
-		// Let's fail fast if Docker is not available
-		if _, err := b.localDocker.ServerVersion(ctx); err != nil {
-			return "", err
-		}
-	}
+// artifactBuilder represents a per artifact builder interface
+type artifactBuilder interface {
+	Build(ctx context.Context, out io.Writer, a *latest.Artifact, tag string) (string, error)
+}
 
+func getArtifactBuilder(b *Builder, a *latest.Artifact) (artifactBuilder, error) {
 	switch {
 	case a.DockerArtifact != nil:
-		return b.buildDocker(ctx, out, a, tag, b.mode)
+		return newDockerBuilder(b), nil
```

Comment:
It there a reason to not follow the other builders and (1) make this a public function and (2) pass in the explicit builder parameters?

Reply:
The problem is that `local` package takes on too much responsibility. It implements both the builder platform (i.e local vs gcb vs cluster) and the per artifact builder (docker, jib, etc) methods. We could extract out a docker package for the single artifact builder and make it consistent with the other single artifact builder packages. But that's a refactor for another time. 🙃

I mainly made this change to be able to write UT for this switch block.

Reply:
the interface is unexported due to the same reason.

Reply:
>  But that's a refactor for another time. 🙃

On second thoughts: https://github.com/GoogleContainerTools/skaffold/pull/4986